src/main/java/frc/robot/sensors/AnalogSelector.java

```diff
@@ -0,0 +1,120 @@
+package frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class AnalogSelector implements Sendable {
+
+    private static final double MODE_ONE_LOW_VOLTAGE = 2.5;
+    private static final double MODE_ONE_HIGH_VOLTAGE = 2.75;
+
+    private static final double MODE_TWO_LOW_VOLTAGE = 2.75;
+    private static final double MODE_TWO_HIGH_VOLTAGE = 3.1;
+
+    private static final double MODE_THREE_LOW_VOLTAGE = 3.1;
+    private static final double MODE_THREE_HIGH_VOLTAGE = 3.5;
+
+    private static final double MODE_FOUR_LOW_VOLTAGE = 3.5;
+    private static final double MODE_FOUR_HIGH_VOLTAGE = 3.75;
+
+    private static final double MODE_FIVE_LOW_VOLTAGE = 3.75;
+    private static final double MODE_FIVE_HIGH_VOLTAGE = 3.95;
+
+    private static final double MODE_SIX_LOW_VOLTAGE = 3.95;
+    private static final double MODE_SIX_HIGH_VOLTAGE = 4.1;
+
+    private static final double MODE_SEVEN_LOW_VOLTAGE = 4.1;
+    private static final double MODE_SEVEN_HIGH_VOLTAGE = 4.2;
+
+    private static final double MODE_EIGHT_LOW_VOLTAGE = 4.2;
+    private static final double MODE_EIGHT_HIGH_VOLTAGE = 4.3;
+    
+    private static final double MODE_NINE_LOW_VOLTAGE = 4.3;
+    private static final double MODE_NINE_HIGH_VOLTAGE = 4.4;
+
+    private static final double MODE_TEN_LOW_VOLTAGE = 4.4;
+    private static final double MODE_TEN_HIGH_VOLTAGE = 4.5;
+
+    private static final double MODE_ELEVEN_LOW_VOLTAGE = 4.5;
+    private static final double MODE_ELEVEN_HIGH_VOLTAGE = 4.6;
+
+    private static final double MODE_TWELVE_LOW_VOLTAGE = 4.6;
+    private static final double MODE_TWELVE_HIGH_VOLTAGE = 5;
+
+    private static final Range[] VOLTAGE_RANGES = {
+        new Range(0, 2.5),
```

Comment:
Do we need this? If so may be worth adding some more constants. Maybe MODE_ZERO?

Reply:
MODE_ZERO makes sense, I'll add that. I wasn't sure what to do with this range as it is never returned as an index.

Reply:
If the physical sensor gets unplugged or damaged, the software might see zero. (Worth testing on the robot by intentionally unplugging the sensor and seeing what happens). Our code should be able to detect this, so a mode zero is a good idea.